zerver/lib/actions.py

```diff
@@ -5050,18 +5050,22 @@ def do_invite_users(user_profile: UserProfile,
                 [], sent_invitations=False)
 
     validated_emails = []  # type: List[str]
-    errors = []  # type: List[Tuple[str, str]]
-    skipped = []  # type: List[Tuple[str, str]]
+    errors = []  # type: List[Tuple[str, str, bool]]
+    skipped = []  # type: List[Tuple[str, str, bool]]
     for email in invitee_emails:
         if email == '':
             continue
         email_error, email_skipped = validate_email(user_profile, email)
         if not (email_error or email_skipped):
             validated_emails.append(email)
         elif email_error:
-            errors.append((email, email_error))
+            errors.append((email, email_error, False))
         elif email_skipped:
-            skipped.append((email, email_skipped))
+            if _("deactivated") in email_skipped:
```

Comment:
This string parsing logic is still not robust.  We should be just expanding the return type of `validate_email` to tell us what the actual situation is.